api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -155,12 +155,32 @@ public Attributes build() {
       return sortAndFilterToAttributes(data.toArray());
     }
 
+    private boolean checkKeyValue(String key, AttributeValue value) {
+      if (key == null || key.length() == 0) {
+        return true;
+      }
+      if (value == null || value.isEmpty()) {
+        int index = data.indexOf(key);
+        if (index == -1) {
+          return true;
+        }
+        // Remove key/value pair
+        data.remove(index);
```

Comment:
This is pretty slow but maybe we don't need to worry.

Reply:
You are right. However, I don't expect the removal of attributes to be a frequent operation.